src/Vehicle/Vehicle.cc

```diff
@@ -840,6 +792,69 @@ void Vehicle::_mavlinkMessageReceived(LinkInterface* link, mavlink_message_t mes
         break;
 #endif
     }
+}
+
+void Vehicle::_mavlinkMessageReceived(LinkInterface* link, mavlink_message_t message)
+{
+    // If the link is already running at Mavlink V2 set our max proto version to it.
+    unsigned mavlinkVersion = _mavlink->getCurrentVersion();
+    if (_maxProtoVersion != mavlinkVersion && mavlinkVersion >= 200) {
+        _maxProtoVersion = mavlinkVersion;
+        qCDebug(VehicleLog) << "Vehicle::_mavlinkMessageReceived Link already running Mavlink v2. Setting _maxProtoVersion" << _maxProtoVersion;
+    }
+
+    if (message.sysid != _id && message.sysid != 0) {
+        // We allow RADIO_STATUS messages which come from a link the vehicle is using to pass through and be handled
+        if (!(message.msgid == MAVLINK_MSG_ID_RADIO_STATUS && _containsLink(link))) {
+            return;
+        }
+    }
+
+    if (!_containsLink(link)) {
+        _addLink(link);
+    }
+
+    //-- Check link status
+    _messagesReceived++;
+    emit messagesReceivedChanged();
+    if(!_heardFrom) {
+        if(message.msgid == MAVLINK_MSG_ID_HEARTBEAT) {
+            _heardFrom = true;
+            _compID = message.compid;
+            _messageSeq = message.seq + 1;
+        }
+    } else {
+        if(_compID == message.compid) {
+            uint16_t seq_received = static_cast<uint16_t>(message.seq);
+            uint16_t packet_lost_count = 0;
+            //-- Account for overflow during packet loss
+            if(seq_received < _messageSeq) {
+                packet_lost_count = (seq_received + 255) - _messageSeq;
+            } else {
+                packet_lost_count = seq_received - _messageSeq;
+            }
+            _messageSeq = message.seq + 1;
+            _messagesLost += packet_lost_count;
+            if(packet_lost_count)
+                emit messagesLostChanged();
+        }
+    }
+
+    // Give the plugin a change to adjust the message contents
+    if (!_firmwarePlugin->adjustIncomingMavlinkMessage(this, &message)) {
+        return;
+    }
+
+    // Give the Core Plugin access to all mavlink traffic
+    if (!_toolbox->corePlugin()->mavlinkMessage(this, link, message)) {
+        return;
+    }
+
+    if (message.compid == MAV_COMP_ID_AUTOPILOT1) {
```

Comment:
this line of code is IMHO - and I believe I correctly understand the MAVlink spec in that matter - eaxctly what one should NOT do, ie., to identify the components type by the standard IDs. 

This type of approach is found repeatedly in the QGC code, but isn't correct (it may work in most practical cases, but isn't guaranteed to work since it's not aligned with the spec)

Reply:
> I believe I correctly understand the MAVlink spec in that matter

Where does it say anything at all about this?